bears/general/AnnotationBear.py

```diff
@@ -0,0 +1,72 @@
+from coalib.parsing.StringProcessing.Core import (search_for,
+                                                  search_in_between,
+                                                  unescaped_search_in_between)
+from coalib.bearlib.languages.LanguageDefinition import LanguageDefinition
+from coalib.bears.LocalBear import LocalBear
+from coalib.results.HiddenResult import HiddenResult
+
+
+class AnnotationBear(LocalBear):
+
+    def run(self, filename, file, language: str, language_family: str):
+        """
+        Finds out all the positions of strings and comments in a file.
+        This does not check however for comments within strings and
+        strings within comments.
+
+        :param language:        Language to be whose annotations are to be
+                                searched.
+        :param language_family: Language family whose annotations are to be
+                                searche.
+
+        :return:                HiddenResult with content conatining
+                                a tuple of tuples of the form
+                                (annotation_start, annotation_end).
```

Comment:
but it belongs in this commit!

Sincerely,

Lasse Schuirmann

contact@viperdev.io
http://viperdev.io/

2016-03-20 17:42 GMT+01:00 Abhay Raizada <notifications@github.com>:

> In bears/general/AnnotationBear.py
> <https://github.com/coala-analyzer/coala-bears/pull/227#discussion_r56767366>
> :
>
> > +class AnnotationBear(LocalBear):
> > +
> > +    def run(self, filename, file, language: str, language_family: str):
> > +        """
> > +        Finds out all the positions of strings and comments in a file.
> > +        This does not check however for comments within strings and
> > +        strings within comments.
> > +
> > +        :param language:        Language to be whose annotations are to be
> > +                                searched.
> > +        :param language_family: Language family whose annotations are to be
> > +                                searche.
> > +
> > +        :return:                HiddenResult with content conatining
> > +                                a tuple of tuples of the form
> > +                                (annotation_start, annotation_end).
>
> @sils1297 <https://github.com/sils1297> i've added that in another commit
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/coala-analyzer/coala-bears/pull/227/files/0bc3f045d70b131ee0774fbf6a49ce3833bfb3ec#r56767366>
>